rotkehlchen/db/dbhandler.py

```diff
@@ -1286,118 +1288,84 @@ def remove_blockchain_accounts(
-            self.msg_aggregator.add_warning(
-                f'Found non-list tokens_list {json_ret} in the DB for {address}.',
-            )
-            return None, last_update_ts
-
-        if len(tokens_list) == 0:
-            return None, last_update_ts
+        last_queried_ts = None
+        cursor.execute(
+            'SELECT key, value FROM accounts_details WHERE account=? AND blockchain=? AND (key=? OR key=?)',  # noqa: E501
+            (address, blockchain.serialize(), ACCOUNTS_DETAILS_LAST_QUERIED_TS, ACCOUNTS_DETAILS_TOKENS),  # noqa: E501
+        )
 
         returned_list = []
-        for x in tokens_list:
-            try:
-                token = EvmToken.from_identifier(x)
-            except (DeserializationError, UnknownAsset):
-                token = None
-            if token is None:
-                self.msg_aggregator.add_warning(
-                    f'Could not deserialize {x} as a token when reading latest '
-                    f'tokens list of {address}',
-                )
-                continue
-
-            returned_list.append(token)
+        for (key, value) in cursor:
+            if key == ACCOUNTS_DETAILS_LAST_QUERIED_TS:
+                last_queried_ts = deserialize_timestamp(value)
+            else:  # should be ACCOUNTS_DETAILS_TOKENS
+                try:
+                    token = EvmToken.from_identifier(value)
+                except (DeserializationError, UnknownAsset):
+                    token = None
+
+                if token is None:
+                    self.msg_aggregator.add_warning(
+                        f'Could not deserialize {value} as a token when reading latest '
+                        f'tokens list of {address}',
+                    )
+                    continue
 
-        return returned_list, last_update_ts
+                returned_list.append(token)
 
-    def _get_address_details_json(self, cursor: 'DBCursor', address: ChecksumEvmAddress) -> Optional[Dict[str, Any]]:  # noqa: E501
-        query = cursor.execute(
-            'SELECT tokens_list, timestamp FROM ethereum_accounts_details WHERE account = ?',
-            (address,),
-        )
-        result = query.fetchall()
-        if len(result) == 0:
-            return None  # no saved entry
-
-        try:
-            json_ret = json.loads(result[0][0])
-        except json.decoder.JSONDecodeError as e:
-            # This should never happen
-            self.msg_aggregator.add_warning(
-                f'Found undecodeable json {result[0][0]} in the DB for {address}.'
-                f'Error: {str(e)}',
-            )
-            return None
+        if len(returned_list) == 0 and last_queried_ts is None:
+            return None, None
 
-        return json_ret
+        return returned_list, last_queried_ts
 
     def save_tokens_for_address(
             self,
             write_cursor: 'DBCursor',
             address: ChecksumEvmAddress,
+            blockchain: SupportedBlockchain,
```

Comment:
Why do you need this? The chain id (and thus the blockchain) is part of `EVMToken`